analyzers/src/SonarAnalyzer.CSharp/Rules/FieldShouldBeReadonly.cs

```diff
@@ -57,7 +57,8 @@ public sealed class FieldShouldBeReadonly : SonarDiagnosticAnalyzer
             SyntaxKind.OrAssignmentExpression,
             SyntaxKind.LeftShiftAssignmentExpression,
             SyntaxKind.RightShiftAssignmentExpression,
-            SyntaxKindEx.CoalesceAssignmentExpression
+            SyntaxKindEx.CoalesceAssignmentExpression,
+            SyntaxKindEx.UnsignedRightShiftAssignmentExpression
         };
```

Comment:
Please make `ReadonlyFieldCollector` and `PartialTypeDeclarationProcessor` classes sealed.